share/enable-completion.sh

```diff
@@ -46,5 +46,5 @@ elif [[ ${ZSH_VERSION-} ]]; then
   # Prepend to `fpath` the path of the directory containing our zsh
   # completion script, so that our completion script will be hooked into the
   # zsh completion system.
-  fpath=("$GIT_SUBREPO_ROOT/share/zsh-completion" "$fpath")
+  fpath=("$GIT_SUBREPO_ROOT/share/zsh-completion" $fpath)
```

Comment:
This patch breaks the tests:
```
 ~/src/git-subrepo PR/511 $ make test test=test/shellcheck.t 
prove  test/shellcheck.t
test/shellcheck.t .. 40/? #   Failed test 'The shell file 'share/enable-completion.sh' passes shellcheck'
#   at test/shellcheck.t line 31.
#     got: '
# In share/enable-completion.sh line 49:
#   fpath=("$GIT_SUBREPO_ROOT/share/zsh-completion" $fpath)
#                                                   ^----^ SC2206: Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.
# 
# For more information:
#   https://www.shellcheck.net/wiki/SC2206 -- Quote to prevent word splitting/g...'
#   expected: ''
test/shellcheck.t .. Failed 1/41 subtests 

Test Summary Report
-------------------
test/shellcheck.t (Wstat: 0 Tests: 41 Failed: 1)
  Failed test:  41
Files=1, Tests=41,  7 wallclock secs ( 0.04 usr  0.02 sys +  6.31 cusr  0.32 csys =  6.69 CPU)
Result: FAIL
make: *** [Makefile:49: test] Error 1
```

The correct way to expand an array variable (in both bash and zsh) is `"${fpath[@]}"`.

I'll apply the PR with that fix.